src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @typedef {Object} VersionNumber
+ * @property {string} major
+ * @property {string} minor
+ * @property {string} patch
+ */
+
+/**
+ * parses a versionNumber string
+ * @returns {VersionNumber}
+ */
+function parseVersionNumber() {
+  const versionElement = document.querySelector(`.link-footer`).textContent
+  const versionNumber = versionElement.match(/v?(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)$/)
+  return versionNumber.groups
+}
+
+
+/**
+ * Replacement onClick handler for Follow button
+ * @param {Event} event 
+ */
+function onClickFollow (event) {
+  event.stopPropagation()
+  event.preventDefault()
+  const username = window.location.pathname.split(`/`).slice(-1)[0]
+  // activate AutoRemoteFollow
+  window.open(`/users/${username}/remote_follow`, `_blank`)
+}
+
+/**
+ * wait for element to appear
+ * @param {string} selector
+ * @param {number} timeout
+ */
+function waitForElement (selector, timeout) {
+  return new Promise((resolve, reject) => {
+    function getElement() {
+      return document.querySelector(selector)
+    }
+
+    const element = getElement()
+    if(element){
+      return resolve(element)
+    }
+
+    const observer = new MutationObserver(() => {
+      const element = getElement()
+      if(element){
+        resolve(element)
+        observer.disconnect()
+      }
+    })
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true
+    })
+
+    window.setTimeout(() => {
+      reject()
+    }, timeout)
+  }) 
+}
+
+/**
+ * Inject replacement onClick handler for Follow button
+ */
+async function injectFollowButton () {
+  try {
+    const followButton = await waitForElement(`.account__header__tabs__buttons button.button`, 20000)
+    followButton.addEventListener(`click`, onClickFollow)
+  } catch {
+    // Follow button failed to appear
+  }
+}
+
+async function init () {
+  let versionNumber
+
+  try {
+    const initialStateObject = JSON.parse(document.getElementById(`initial-state`).innerHTML)
+    const version = initialStateObject?.meta?.version
+    if(!initialStateObject || !version){
+      // not a Mastodon server
+      return
+    }
+
+    versionNumber = parseVersionNumber(version)
+  } catch {
+    return
+  }
+
+  
+  if(Number.parseInt(versionNumber.major) >= 4){
```

Comment:
Again hmm, the point stands of how useful it is to check the version and ignore < v4 versions? Is this an advantage or rather a potential thing that could break stuff?

Reply:
I guess that's fair. The CSS selector we use to identify the v4 Follow button already essentially helps us distinguish v4 Mastodon instances so there's no need to do a prior check.